Setup github action to sync external plugins

urwrstkn8mare opened the pull request:
I just wrote this up after looking at various GitHub actions documentation and combining some GitHub actions. Haven't gotten the time to test it yet but I think it would be easier for @UncleGoogle to test this anyway. Please have a look and will appreciate any suggestions. 

PS. Then we can use the same workflow that is used for all plugins (internally maintained or otherwise) to create releases and update `current_version.json`, etc. In fact, I suggest we create another custom GitHub action that abstracts some of the repetitive stuff out of said workflow. 

PSS. github codespaces is pretty cool

urwrstkn8mare committed:
add action.yml

urwrstkn8mare committed:
update README

AndrewDWhite commented on README.md:
```diff
@@ -1 +1,20 @@
+      - uses: GOG-Nebula/syncer@v0
+        with:
+          upstream: owner/repo
+          reviewers: username1,username2
```
You will want to update the reviewers. Is there a way to just add a group of them without calling them all out?

UncleGoogle replied:
@AndrewDWhite  -- +1 your concern 

I can see there is `team-reviewer` in peter-evans/create-pull-request@v4

so we could have one "members" team, or multiple teams for different corners in the future.

But it has an issue: https://github.com/peter-evans/create-pull-request/issues/155#issuecomment-611904487

so that we would have to use artificial github-account like FriendsOfGalaxyBot. But probably don't have to put its secret to each repo - only to this github action. Probably. Maybe. I don't know...

---------

Alternatively we can just list all repo members with other step and pass the resutl to create PR step.

something similar (but simpler) like this: https://github.com/RITct/members-list-action

---

But that is quality life improvement we can probably start without.

urwrstkn8mare replied:
> so that we would have to use artificial github-account like FriendsOfGalaxyBot. But probably don't have to put its secret to each repo - only to this github action. Probably. Maybe. I don't know...

Yeah just seeing that, its pretty unfortunate not sure why thats a thing. 

> Alternatively we can just list all repo members with other step and pass the resutl to create PR step.

I think this is a great workaround. We could even automatically remove the upstream owner from the list.

UncleGoogle commented on action.yml:
```diff
@@ -0,0 +1,51 @@
+      uses: peter-evans/create-pull-request@v4
+      with:
+        base: ${{ github.repository }}
+        branch: pr/sync
```
maybe just "sync" or "autosync"?

urwrstkn8mare replied:
yeah that works, I just arbitrarily chose something. Actually, I kind of like the `/` convention to give a clue as to what the branch is for. For example, `hotfix/appid` or `feature/ui`. Actually now that I think of it I can't think of uses for `pr/` so maybe `auto/sync`? In case we have another action that automatically creates a branch? Honestly tho, `autosync` is perfectly fine, feel free to make that change or if you have another alternative suggest it and I'll make it.
```suggestion
        branch: autosync
```

UncleGoogle commented on action.yml:
```diff
@@ -0,0 +1,51 @@
+        labels: auto sync
+        reviewers: ${{ inputs.reviewers }}
+    - name: Check outputs
+      if: ${{ steps.cpr.outputs.pull-request-number }}
```
what does it check in practice? If the PR was created sucessfully?

UncleGoogle commented on README.md:
````diff
@@ -1 +1,20 @@
+          upstream: owner/repo
+          reviewers: username1,username2
+```
+***NOTE: should not be used in an integration (eg. cron value is just a random value)***
````
What do you mean by "should no be used in an integration"? Is it something beyond a title "## Example Usage (workflow)"?

urwrstkn8mare replied:
I just meant that cause I took the example off the internet. The example was mainly to show its parameters and how it would be used in a workflow. For example, the cron value I haven't even bothered to check what it means. We could and should update the example to something that closely matches our usage after we test it.

UncleGoogle commented on README.md:
```diff
@@ -1 +1,20 @@
+  sync:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: GOG-Nebula/syncer@v0
```
I dont see v0 specified in action.yml. Could you add?

urwrstkn8mare replied:
I don't think the version is specified in `action.yml`. It is defined as a tag on a commit. But I realise just using the branch name is better for now.

UncleGoogle commented:
@urwrstkn8mare awesome, brilliant! I'm excited :rocket: 

I'll test it for sure. Sadly - won't test in a next few days. Tuesday is more realistic, so if you have time - feel free to close PR in hubmle bundle fork and setup synchronizer there

BTW If I understand github action versioning correctly I can test it before merging by specyfing `syncer@branch_name` ?

urwrstkn8mare commented:
Oh I almost forgot I have my own plugin to test with, although it isn't in built-in search I hope you guys don't mind me forking it to the org to at least test.

urwrstkn8mare committed:
add ref input + update README

urwrstkn8mare commented:
For whatever reason the workflow isn't running so if someone is able to check what's going on I'd greatly appreciate (see https://github.com/GOG-Nebula/galaxy-riot-integration)

AndrewDWhite commented:
> For whatever reason the workflow isn't running so if someone is able to check what's going on I'd greatly appreciate (see https://github.com/GOG-Nebula/galaxy-riot-integration)

The yml is not valid. If you would like a more detailed answer let me know and I can look at it more later when I feel better.
 https://github.com/GOG-Nebula/galaxy-riot-integration/actions/runs/2399870420
 https://github.com/GOG-Nebula/syncer/blob/8d3eacaa7733c987da341392202c8aff039f72c1/action.yml#L24

An example of how you can check this locally.

$ python -c 'import yaml, sys; print(yaml.safe_load(sys.stdin))' < action.yml   Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "C:\Users\andyn\AppData\Local\Programs\Python\Python37-32\lib\site-packages\yaml\__init__.py", line 162, in safe_load
    return load(stream, SafeLoader)
  File "C:\Users\andyn\AppData\Local\Programs\Python\Python37-32\lib\site-packages\yaml\__init__.py", line 114, in load
    return loader.get_single_data()
  File "C:\Users\andyn\AppData\Local\Programs\Python\Python37-32\lib\site-packages\yaml\constructor.py", line 41, in get_single_data
    node = self.get_single_node()
  File "C:\Users\andyn\AppData\Local\Programs\Python\Python37-32\lib\site-packages\yaml\composer.py", line 36, in get_single_node
    document = self.compose_document()
  File "C:\Users\andyn\AppData\Local\Programs\Python\Python37-32\lib\site-packages\yaml\composer.py", line 55, in compose_document
    node = self.compose_node(None, None)
  File "C:\Users\andyn\AppData\Local\Programs\Python\Python37-32\lib\site-packages\yaml\composer.py", line 84, in compose_node
    node = self.compose_mapping_node(anchor)
  File "C:\Users\andyn\AppData\Local\Programs\Python\Python37-32\lib\site-packages\yaml\composer.py", line 133, in compose_mapping_node
    item_value = self.compose_node(node, item_key)
  File "C:\Users\andyn\AppData\Local\Programs\Python\Python37-32\lib\site-packages\yaml\composer.py", line 84, in compose_node
    node = self.compose_mapping_node(anchor)
  File "C:\Users\andyn\AppData\Local\Programs\Python\Python37-32\lib\site-packages\yaml\composer.py", line 133, in compose_mapping_node
    item_value = self.compose_node(node, item_key)
  File "C:\Users\andyn\AppData\Local\Programs\Python\Python37-32\lib\site-packages\yaml\composer.py", line 82, in compose_node
    node = self.compose_sequence_node(anchor)
  File "C:\Users\andyn\AppData\Local\Programs\Python\Python37-32\lib\site-packages\yaml\composer.py", line 111, in compose_sequence_node
    node.value.append(self.compose_node(node, index))
  File "C:\Users\andyn\AppData\Local\Programs\Python\Python37-32\lib\site-packages\yaml\composer.py", line 84, in compose_node
    node = self.compose_mapping_node(anchor)
  File "C:\Users\andyn\AppData\Local\Programs\Python\Python37-32\lib\site-packages\yaml\composer.py", line 127, in compose_mapping_node
    while not self.check_event(MappingEndEvent):
  File "C:\Users\andyn\AppData\Local\Programs\Python\Python37-32\lib\site-packages\yaml\parser.py", line 98, in check_event
    self.current_event = self.state()
  File "C:\Users\andyn\AppData\Local\Programs\Python\Python37-32\lib\site-packages\yaml\parser.py", line 428, in parse_block_mapping_key
    if self.check_token(KeyToken):
  File "C:\Users\andyn\AppData\Local\Programs\Python\Python37-32\lib\site-packages\yaml\scanner.py", line 116, in check_token
    self.fetch_more_tokens()
  File "C:\Users\andyn\AppData\Local\Programs\Python\Python37-32\lib\site-packages\yaml\scanner.py", line 223, in fetch_more_tokens
    return self.fetch_value()
  File "C:\Users\andyn\AppData\Local\Programs\Python\Python37-32\lib\site-packages\yaml\scanner.py", line 579, in fetch_value
    self.get_mark())
yaml.scanner.ScannerError: mapping values are not allowed here
  in "<stdin>", line 24, column 33

urwrstkn8mare committed:
hotfix: invalid action.yml

urwrstkn8mare committed:
hotfix: missing property -

urwrstkn8mare committed:
hotfix: cpr  prop invalid + unnecessary

urwrstkn8mare commented:
Ok so there is an issue with this. I'm not sure whats wrong but I gotta fix it later. I think the create-pull-request action doesn't work as I thought it did. If you guys have any suggestions please feel free.

urwrstkn8mare committed:
add files from actions/typescript-action

urwrstkn8mare committed:
wrote the action - not tested

urwrstkn8mare committed:
make token required

urwrstkn8mare committed:
add console log for debugging

urwrstkn8mare committed:
replaced console.log with core.info

urwrstkn8mare commented:
Now I rewrote it as a typescript GitHub actions which theoretically is more efficient and straightforward. But now the jobs are returning this error: https://github.com/GOG-Nebula/galaxy-riot-integration/runs/6637836565?check_suite_focus=true. I don't know how to trace the error so this is quite annoying. I've setup a test branch in my personal repo and forked the master branch here. So its ready to test. Any help would be greatly appreciated (also feel free to directly commit to the master branch of the forked riot plugin as its just there for testing for now).

urwrstkn8mare committed:
forgot to package action smh

urwrstkn8mare committed:
decoded manifest content

urwrstkn8mare committed:
forgot to package agai

urwrstkn8mare commented:
oooooooooooooh my god it works. see https://github.com/GOG-Nebula/galaxy-riot-integration/pull/1 (don't actually merge lol). The commits are messy but maybe its time to try out that squash and merge functionality. Also this could still use alot of polishing:
- [ ] better logging in the action
- [ ] update the readme
- [ ] fix this repo's workflows (incl making them use yarn instead of npm)
- [ ] add tests (maybe not worth it)
- [ ] improve code readablitiy

So I'd appreciate any help with that as I shall now sleep. 

EDIT: moved to readme as it is **not** something that needs to be done ASAP.

urwrstkn8mare committed:
update name

urwrstkn8mare committed:
add todo to readme

urwrstkn8mare committed:
fix: use yarn instead of npm in github workflows

urwrstkn8mare committed:
udocs: update readme

urwrstkn8mare committed:
yarn package

urwrstkn8mare committed:
update readme

Mixaill commented on README.md:
```diff
@@ -1 +1,36 @@
+  sync:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: GOG-Nebula/action-sync-fork@feature/initial
```
do not forget to switch branch to `main` before merge

urwrstkn8mare replied:
Oh yeah thanks!
```suggestion
      - uses: GOG-Nebula/action-sync-fork@main
```

urwrstkn8mare committed:
Update README.md

Co-authored-by: Mikhail Paulyshka <me@mixaill.net>

urwrstkn8mare commented:
@UncleGoogle @AndrewDWhite if you could have a look please in your own time.

urwrstkn8mare committed:
update usage title - README

UncleGoogle commented:
I'll try to test&review tonight. Definitely squash is good in that case. Merge/rebase only if each commit brings atomic, relevant change.

> add tests (maybe not worth it)

If you don't add tests now, probably nobody will add them later :) I would be good to have test first approach for all code we do. But it a little less value _after_ fact that you already tested it manually. Tests (especially TDD) saves time during developement, to don't have to spend too much time on manual testing. And of course would be useful in future changes.

UncleGoogle commented on README.md:
```diff
@@ -1 +1,36 @@
-# syncer
\ No newline at end of file
+# action-sync-fork
```
```suggestion
# actions-sync-fork
```

UncleGoogle commented on package.json:
```diff
@@ -0,0 +1,40 @@
+{
+  "name": "sync-fork",
+  "version": "0.0.0",
```
```suggestion
  "version": "0.1.0",
```

suggestion for semantic versioning-compilant

UncleGoogle commented on src/main.ts:
```diff
@@ -0,0 +1,75 @@
+        owner: upstream_owner,
+        repo: upstream_repo,
+        ref: upstream_branch,
+        path: 'src/manifest.json'
```
not every repo has this structure eg https://github.com/Mixaill/galaxy-integration-wargaming

We can use optional parameter to this action or do some magic like here: https://github.com/FriendsOfGalaxy/galaxy-integrations-synchronizer/blob/cbf572ed1639813d90596ba37257425544550da9/scripts.py#L192

urwrstkn8mare replied:
Hmm I see that is definitely something we should consider in the future. But I was also thinking we enforce a light repository structure (e.g. plugin files in `src`) as it is *imo* cleaner and easier to package. And by enforce I don't mean denying plugins or anything like that, just could be as simple as altering the repo structure in the fork. Just an idea tho. Otherwise making the path to the manifest can be done via an optional parameter as you said. If so let me know and I'll do it next or you could try it yourself as I will likely not be able to work on this for the next few days. Sorry.

UncleGoogle commented on src/main.ts:
```diff
@@ -0,0 +1,75 @@
+    (await oktokit.rest.repos.get({owner: upstream_owner, repo: upstream_repo}))
+      .data.default_branch
+  const reviewers = core.getInput('reviewers').split(',')
+  const base_branch = (await oktokit.rest.repos.get({owner, repo})).data
+    .default_branch
```
suggestion

an idea taken from wei/pull:

our fork could have default_branch set to another branch with custom README (like those readme's in FoG forks that redirects to upstream repo for issues and support) and workflows to avoid conflicts with main branch from upstream. Other files could be even removed completely on that branch.

I'm not sure it is the best idea to solve problem of potential conflicts, just one solution.

For now it is pre-mature care probably, so I'm OK with syncing `upstream_branch` with `default_branch` of fork.

urwrstkn8mare replied:
Oh that is quite easy to implement so we should probably just implement it when its needed.

urwrstkn8mare replied:
But if it should be done now, then let me know and I'll do that next or you can add in yourself if you want because I will not be back for a few days. Sorry.

UncleGoogle commented on src/main.ts:
```diff
@@ -0,0 +1,75 @@
+      const {data: base_manifest} = await oktokit.rest.repos.getContent({
+        owner,
+        repo,
+        path: 'src/manifest.json'
```
the same as above

UncleGoogle commented on src/main.ts:
```diff
@@ -0,0 +1,75 @@
+        repo,
+        path: 'src/manifest.json'
+      })
+      if ('content' in base_manifest && 'content' in upstream_manifest) {
```
what means the other case? I mean the `else` -- would it be an unexpected response of oktokit? Is logging more appropriate or `core.setFailed(error)`?

urwrstkn8mare replied:
See https://github.com/GOG-Nebula/action-sync-fork/pull/1#issuecomment-1146227922

UncleGoogle commented on src/main.ts:
```diff
@@ -0,0 +1,75 @@
+            head: upstream_owner + ':' + upstream_branch,
+            base: base_branch,
+            maintainer_can_modify: false,
+            title: 'New version from upstream'
```
suggestion: can include `upstream_version`

urwrstkn8mare replied:
I just put it in the body as the PR I think will just be merging the latest commits from the upstream to the base anyway and as there can only be *one* `autosync` PR, I didn't feel the need to add it. However feel free to suggests a commit/change to add `upstream_version` if that would be better.

UncleGoogle reviewed (approved):
@urwrstkn8mare I don't know ts, but I've checked the logic.

I've put some comments, but only one thing is "major" -- the `manifest.json` lookup.

However we can improve to support not standard manifest.json location in next iteration, that's why :heavy_check_mark: from me

AndrewDWhite commented:
I know basically nothing about yarn but a nearly 9k length file in dist as a versioned file doesn't sound like it is going to be easy to maintain.

UncleGoogle commented:
@AndrewDWhite it is a generated lock file - not something you have to manually maintain. It pins all elements of all the dependency tree specified in `package.json`. Similar to pipenv or poetry lock's.

That's way you're sure that each time the app is installed, it should be exactly the same. So we shouldn't be affected by similar issues to this one without manual update of dependencies
https://www.reddit.com/r/Python/comments/uwhzkj/i_think_the_ctx_package_on_pypi_has_been_hacked/

UncleGoogle commented on README.md:
```diff
@@ -1 +1,36 @@
+  sync:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: GOG-Nebula/action-sync-fork@main
```
```suggestion
      - uses: GOG-Nebula/actions-sync-fork@main
```

or rename the repo if that is better

UncleGoogle commented on README.md:
```diff
@@ -1 +1,36 @@
+
+## Example Usage
+
+[`.github/workflows/sync.yml`](https://github.com/GOG-Nebula/galaxy-riot-integration/blob/9da15bdfd5f30f2aa6e887491d7f53d0853adeae/.github/workflows/sync.yml)
```
outdated content. Lets remove and keep only the example below

urwrstkn8mare replied:
done

UncleGoogle reviewed (changes requested):
I made some tests.

No logs /errors when I made a typo here:

https://github.com/GOG-Nebula/galaxy-integration-humblebundle/runs/6699158928?check_suite_focus=true

after I fixed it, there is still no results:

https://github.com/GOG-Nebula/galaxy-integration-humblebundle/runs/6699204856?check_suite_focus=true

What is wrong in my setup?

urwrstkn8mare commented:
> I know basically nothing about yarn but a nearly 9k length file in dist as a versioned file doesn't sound like it is going to be easy to maintain.

Oh yeah and the files in dist are built from src because github actions requires dependency-less JavaScript available on the repo. And yeah the yarn.lock file @UncleGoogle has explained.

urwrstkn8mare committed:
Update README.md

Co-authored-by: Mesco <mieszkoziemowit@gmail.com>

urwrstkn8mare commented:
> I made some tests.
> 
> No logs /errors when I made a typo here:
> 
> https://github.com/GOG-Nebula/galaxy-integration-humblebundle/runs/6699158928?check_suite_focus=true
> 
> after I fixed it, there is still no results:
> 
> https://github.com/GOG-Nebula/galaxy-integration-humblebundle/runs/6699204856?check_suite_focus=true
> 
> What is wrong in my setup?

I'm not quite sure actually. It was working with the riot repo. I guess this is a sign to make the action log and implement tests before merging.

AndrewDWhite commented:
> > I know basically nothing about yarn but a nearly 9k length file in dist as a versioned file doesn't sound like it is going to be easy to maintain.
> 
> Oh yeah and the files in dist are built from src because github actions requires dependency-less JavaScript available on the repo. And yeah the yarn.lock file @UncleGoogle has explained.

I guess what I am indirectly thinking about and asking is this. Is is easier, i.e. a built in function, to maintain a current hash of those files, just like the lockfile does, instead of those files if the content is meaningless to humans, auto generated, and only the presence of a difference in what is generated matters?

urwrstkn8mare committed:
createepo class

urwrstkn8mare committed:
improved logging + readability

urwrstkn8mare committed:
improve pulls search logging

urwrstkn8mare committed:
refactor class name

urwrstkn8mare committed:
yarn package

urwrstkn8mare committed:
added some

urwrstkn8mare committed:
add log to check if debug enabled

urwrstkn8mare committed:
yarn package

urwrstkn8mare committed:
hotfix: use .number instead of .id (pulls)

urwrstkn8mare committed:
yarn package

urwrstkn8mare committed:
add repo to pulls query

urwrstkn8mare committed:
yarn package

urwrstkn8mare committed:
add awaits

urwrstkn8mare committed:
try swapping base and upstream

urwrstkn8mare committed:
add debug log to check

urwrstkn8mare committed:
a typo :(

urwrstkn8mare committed:
another typo 🤬

urwrstkn8mare committed:
emphasise example and remove note in README

urwrstkn8mare committed:
Update package.json

Co-authored-by: Mesco <mieszkoziemowit@gmail.com>

urwrstkn8mare committed:
rename repo

urwrstkn8mare commented:
@UncleGoogle I've got it working now, see: https://github.com/GOG-Nebula/galaxy-integration-humblebundle/pull/4

I've also made it readable and added plenty of logging while I was at it. (and renamed the repo)

So personally I think it is ready to merge but please have a look and let me know your thoughts.

> I guess what I am indirectly thinking about and asking is this. Is is easier, i.e. a built in function, to maintain a current hash of those files, just like the lockfile does, instead of those files if the content is meaningless to humans, auto generated, and only the presence of a difference in what is generated matters?

Sorry I'm not sure what you mean here. @AndrewDWhite could you elaborate?

AndrewDWhite commented:
> @UncleGoogle I've got it working now, see: [GOG-Nebula/galaxy-integration-humblebundle#4](https://github.com/GOG-Nebula/galaxy-integration-humblebundle/pull/4)
> 
> I've also made it readable and added plenty of logging while I was at it. (and renamed the repo)
> 
> So personally I think it is ready to merge but please have a look and let me know your thoughts.
> 
> > I guess what I am indirectly thinking about and asking is this. Is is easier, i.e. a built in function, to maintain a current hash of those files, just like the lockfile does, instead of those files if the content is meaningless to humans, auto generated, and only the presence of a difference in what is generated matters?
> 
> Sorry I'm not sure what you mean here. @AndrewDWhite could you elaborate?

Since some of the files are simply automatically generated; instead of the file being checked into the repo we would store a file that looks like the lock files for the dependencies. (There is probably something that already exists for this since it effectively duplicates the lock files but instead of dependencies it does it for the generated files.)

Here is an example entry for `dist/index.js` .
`"@dist/index.js@":
  integrity sha512-df52a38d211d06913d101056ce43640391dee11abe9a5c379aeda7167210679850512f0b99e8e7951dbd44f5ce9336fa10350d917023788fde33085a8140b634`

In the process, you would hash and append all of the other generated files cryptographically secure hashes into the file.
`"@dist/index.js@":
  integrity sha512-df52a38d211d06913d101056ce43640391dee11abe9a5c379aeda7167210679850512f0b99e8e7951dbd44f5ce9336fa10350d917023788fde33085a8140b634`

`"@dist/index.js.map@":
integrity sha512-8ad74f602c2048130f1bc29dd18e61313541399f384a557b130fd76043a9cfebb9fbb7ed00fa120a7f6aec96bf37d9b8179925fc5f7a06619f7f8c8038f26136`
And so on filling the file with the hashes for the other generated files. (Note this will have all automatically generated files' hashes and not just changes.)

In the processing what you would do is generate these files normally using the automatic process. Instead of checking the hash against the hash of the file directly stored by git, instead you can check the hash against the corresponding value in the file. The advantage here is that there is less code in the repo for anyone to review for changes, and changes would be updated in that file and still validated.

![image](https://user-images.githubusercontent.com/972757/171965034-c33965ef-20da-4996-8bb0-1f3182079eb6.png)


An example of how to get the example hash value is the following.
`$ sha512sum.exe index.js
df52a38d211d06913d101056ce43640391dee11abe9a5c379aeda7167210679850512f0b99e8e7951dbd44f5ce9336fa10350d917023788fde33085a8140b634 *index.js`

urwrstkn8mare commented:
Oh I see. Correct me if I'm wrong but doesn't github already only check in changed files?

urwrstkn8mare commented:
Oh @AndrewDWhite I think I understand what you mean, storing the hashes of `dist/` instead of the files itself. Unfortunately, for github actions, the `dist/` must be checked in as that is where the files for the Github action are downloaded from on run. Normally in a JS project, `dist/` isn't checked in at all.

UncleGoogle commented:
Looks like github actions API does not support typescript natievly (like browsers) so that people advice commiting dist each time: https://github.com/actions/typescript-action#publish-to-a-distribution-branch

If you want to takle a risk of not being able to review minified raw dist files, what about having that check in pipeline? Basically either generate and commit it another commit by bot or guard in CI (simpler option): fail pipeline if committed `dist` differs from the generated `dist` during the CI step.

urwrstkn8mare commented:
> fail pipeline if committed dist differs from the generated dist during the CI step.

Yep there's alr a workflow doing that.

AndrewDWhite reviewed (approved):
Manually reviewed all but the following automatically generated files:
- dist/index.js
- dist/index.js.map
- dist/licenses.txt
- dist/sourcemap-register.js

- yarn.lock

If the first four files end up being unmanageable with this approach we can look at adding the processing steps described in the discussion into the workflow.

urwrstkn8mare commented:
> Manually reviewed all but the following automatically generated files:
> 
> * dist/index.js
> * dist/index.js.map
> * dist/licenses.txt
> * dist/sourcemap-register.js
> * yarn.lock
> 
> If the first four files end up being unmanageable with this approach we can look at adding the processing steps described in the discussion into the workflow.

For the `dist/` files its ok cause if all the checks pass that means that it is what it should be. As for the yarn.lock, I don't think other JS projects do anything to deal with it when reviewing changes.

urwrstkn8mare committed:
add contributing guidelines
